pipeline/rh-upi-pipeline-executor.groovy

```diff
@@ -178,49 +230,31 @@ node("magna006") {
     }
 
 
-    def testSuites = fetchSuites(scenario, sharedLib)
+    def testSuites = fetchStages(scenario, sharedLib)
     println(testSuites)
     testSuites.each { k, v ->
         logDir = "/ceph/cephci-jenkins/upi/${env.BUILD_NUMBER}/${k}/"
-        def testJobStages = fetchStages(
+        def testJobStages = fetchSuites(
             v, logDir, testResults, k, clusterConf, rhBuild, platform, sharedLib
         )
 
+        println("abort_on_failure")
+        println(v.abort_on_failure)
+        abort_on_failure = v.abort_on_failure
         parallel testJobStages
-        println(testResults)
-    }
-
-    stage("publishResults") {
-        println("Publish results for ${scenario}")
-        sharedEmailLib = load(
-            "${env.WORKSPACE}/pipeline/vars/bm_send_test_reports.groovy"
-        )
-        println(testResults)
-
-        yamlData = readYaml(
-            file: "/ceph/cephci-jenkins/latest-rhceph-container-info/${params.rhcephVersion}.yaml"
-        )
-        println(yamlData)
-        println(sharedEmailLib)
-        sharedEmailLib.sendEMail(testResults, rhBuild, scenarioName, yamlData["tier-0"])
-    }
-
-    stage("PostBuildAction") {
-        def nextScenario = scenario.toInteger()+1
-        def testStages = fetchSuites(nextScenario.toString(), sharedLib)
-
-        if (! testStages) {
-            println("Completed all stages")
-            return
+        def stageResult = ""
+        testResults[k].entrySet().each{
+            if("${abort_on_failure}" == "true" && "${it.value["Result"]}" == "FAIL"){
+                println("ABORT all Stages")
+                try { error("${testResults[k]} failed so aborting all other stages") }
+            finally{
+            publishResultsStage(scenario,testResults, rhBuild, scenarioName)
+            postBuildActionStage(sharedLib,scenario)
+                }
+            }
         }
-
-        build([
-            wait: false,
-            job: "rh-upi-pipeline-executor",
-            parameters: [
-                string(name: 'rhcephVersion', value: params.rhcephVersion),
-                string(name: 'scenarioId', value: nextScenario.toString())
-            ]
-        ])
+        println(testResults)
     }
-}
+    publishResultsStage(scenario,testResults, rhBuild, scenarioName)
+    postBuildActionStage(sharedLib,scenario)
+}
```

Comment:
A new line at the end of the file is missing.

Reply:
This file merged accidentally.Removed it